### DISCUSSION TITLE
Computing derivatives with respect to cartesian coordinates

### DISCUSSION THREAD

**Discussion by aglgit** [deepmodeling/deepmd-kit#31] [Q&A] [answered]
Your article on DeepMD-kit describes how to build the descriptors and how
to obtain the forces, however I am unsure how you take the derivatives
of the descriptors with respect to the cartesian coordinates.
These seem like rather complicated expressions due to the presence of the rotation
matrix and unit vectors, however the omission of this in your papers leads me to believe that this is not as complicated as I would guess. Do you have this somewhere, or can you offer some guidance on how to obtain them?

**Comment by deepmodeling** [comment, marked as answer]
Please check the implementation in 
source/lib/include/ComputeDescriptor.h
